### PR TITLE
fix: [CDS-87114]: filtering clusters from linkedcluster for environment

### PR DESCRIPTION
**Pull request by sahithikolichala**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

#### [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

Problem: 
We were showing all the clusters , this would result into execution error
Solution:
Filtered With the linkedClusters
https://github.com/harness/harness-core-ui/assets/57372986/d94b6739-f0d4-4944-9a04-c890fc5d9ae9

### CODE REVIEW

**Review by cjlee01 — APPROVED** [review]

File: src/modules/70-pipeline/components/FormMultiTypeGitOpsClusterField/Utils.tsx [hunk 1]
```diff
@@ -103,16 +116,24 @@ export function getReferenceFieldProps({
...
-          if (responseData?.data?.content) {
-            const content = responseData?.data?.content?.map(item => ({
-              ...item,
-              identifier: `${item.identifier}${DELIMITER}${item.agentIdentifier}`,
-              record: {
-                ...item,
-                ...orgAndProj
+          if (responseData?.data?.content && responseData?.data?.content.length) {
+            const content = []
+            for (const item of responseData.data.content) {
```
Comment (abhinavrastogi-harness) [inline src/modules/70-pipeline/components/FormMultiTypeGitOpsClusterField/Utils.tsx:121]: why not use map-reduce here instead?

File: src/modules/70-pipeline/components/FormMultiTypeGitOpsClusterField/Utils.tsx [hunk 2]
```diff
@@ -103,16 +116,24 @@ export function getReferenceFieldProps({
...
+                  record: {
+                    ...item,
+                    ...orgAndProj
+                  }
+                })
               }
-            }))
-            setPagedClusterData(responseData)
+            }
+            setPagedClusterData(content as ResponsePageClusterFromGitops)
```
Comment (abhinavrastogi-harness) [inline src/modules/70-pipeline/components/FormMultiTypeGitOpsClusterField/Utils.tsx:136]: it's better to set the type on initialisation on line 120 instead

File: src/modules/70-pipeline/components/FormMultiTypeGitOpsClusterField/Utils.tsx (hunk 1 above)
Comment (sahithikolichala) [reply src/modules/70-pipeline/components/FormMultiTypeGitOpsClusterField/Utils.tsx:121]: will refactor as follow up

**Review by cjlee01 — APPROVED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

File: src/modules/70-pipeline/components/FormMultiTypeGitOpsClusterField/__tests__/FormMultiTypeEntitySelector.test.tsx [hunk 3]
```diff
@@ -64,7 +68,7 @@ describe('FormMultiTypeConnectorField tests', () => {
     fireEvent.click(getByText('account'))
 
     await waitFor(() => {
-      expect(getByTestId(dialog, 'cluster16')).toBeInTheDocument()
+      expect(getByTestId(dialog, 'incluster')).toBeInTheDocument()
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/FormMultiTypeGitOpsClusterField/__tests__/FormMultiTypeEntitySelector.test.tsx:71]: Should not we check for proper value of the field?

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/70-pipeline/components/FormMultiTypeGitOpsClusterField/__tests__/FormMultiTypeEntitySelector.test.tsx (hunk 3 above)
Comment (sahithikolichala) [reply src/modules/70-pipeline/components/FormMultiTypeGitOpsClusterField/__tests__/FormMultiTypeEntitySelector.test.tsx:71]: for this element , testId is the identifier
so i am checking on that

**Pull request merged**